gateway/src/test/java/io/camunda/zeebe/gateway/interceptors/impl/IdentityInterceptorTest.java

```diff
@@ -108,8 +115,79 @@ public void validTokenIsAccepted() {
             });
 
     // then
-    assertThat(closeStatusCapturingServerCall.closeStatus)
-        .hasValueSatisfying(status -> assertThat(status.getCode()).isEqualTo(Status.OK.getCode()));
+    assertThat(closeStatusCapturingServerCall.closeStatus).hasValue(Status.OK);
+  }
+
+  @Test
+  public void addsAuthorizedTenantsToContext() {
+    // given
+    final Identity identity = mock(Identity.class, RETURNS_DEEP_STUBS);
+    when(identity.tenants().forToken(anyString()))
+        .thenReturn(List.of(new Tenant("tenant-a", "A"), new Tenant("tenant-b", "B")))
+        .thenReturn(List.of(new Tenant("tenant-c", "C")));
+    final var capturingServerCall = new CloseStatusCapturingServerCall();
+
+    // when
+    final var interceptor = new IdentityInterceptor(identity, multiTenancy.setEnabled(true));
+    interceptor.interceptCall(
+        capturingServerCall,
+        createAuthHeader(),
+        (call, headers) -> {
+          // then
+          assertAuthorizedTenants()
+              .describedAs("Expect that the authorized tenants is stored in the current Context")
+              .contains("tenant-a", "tenant-b");
+          call.close(Status.OK, headers);
+          return null;
+        });
+
+    // when a second call is intercepted, the authorized tenants should be updated
+    interceptor.interceptCall(
+        capturingServerCall,
+        createAuthHeader(),
+        (call, headers) -> {
+          // then
+          assertAuthorizedTenants()
+              .describedAs(
+                  "Expect that the authorized tenants is different in another request's Context")
+              .contains("tenant-c");
+          call.close(Status.OK, headers);
+          return null;
+        });
+
+    // then
+    assertThat(capturingServerCall.closeStatus).hasValue(Status.OK);
+    assertAuthorizedTenants()
+        .describedAs("Expect that the authorized tenants is not available outside of a call")
+        .isNull();
+  }
+
+  @Test
+  public void doesNotAddAuthorizedTenantsToContextWhenMultiTenancyDisabled() {
```

Comment:
:thinking: We can consider adding the `<default>` tenant id to the gRPC context instead when MT is disabled, that way, the JWT provided to the BrokerRequest will always be populated with tenant ids.

Reply:
I took a different route. It's not dependent on Identity (and the IdentityInterceptor), but also when another custom tenant authorization provider is configured, we should set the default tenant when MT is disabled.